craft_parts/parts.py

```diff
@@ -0,0 +1,243 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Definitions and helpers to handle parts."""
+
+import copy
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Set
+
+from craft_parts import errors
+from craft_parts.dirs import ProjectDirs
+from craft_parts.steps import Step
+
+
+class Part:
+    """Each of the components used in the project specification.
+
+    During the craft-parts lifecycle each part is processed through
+    different steps in order to obtain its final artifacts. The Part
+    class holds the part specification data and additional configuration
+    information used during step processing.
+
+    :param name: The part name.
+    :param data: A dictionary containing the part properties.
+    :param project_dirs: The project work directories.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        data: Dict[str, Any],
+        *,
+        project_dirs: ProjectDirs = None,
+    ):
+        if not project_dirs:
+            project_dirs = ProjectDirs()
+
+        self._name = name
+        self._data = data
+        self._dirs = project_dirs
+        self._part_dir = project_dirs.parts_dir / name
+
+    def __repr__(self):
+        return f"Part({self.name!r})"
+
+    @property
+    def name(self) -> str:
+        """Return the part name."""
+        return self._name
+
+    @property
+    def properties(self) -> Dict[str, Any]:
+        """Return the part properties."""
+        return self._data.copy()
+
+    @property
+    def parts_dir(self) -> Path:
```

Comment:
why is this necessary? Shouldn't this be handled a layer above?

Reply:
Yes, this shouldn't be here. It looks like a leftover from a previous refactoring (and it's not used in the rest of the PoC code).